### PR TITLE
fix(nextjs): make next build package manager agnostic

### LINKED ISSUE #18653: @nx/next 16.7.0 - "Invalid project directory provided, no such directory"
### Current Behavior

We've just updated our workspace to 16.7.0 using `nx migrate` as a resolution to [this issue](https://github.com/nrwl/nx/issues/17987). That all went well.

Upon completing the update, we've found that we are unable to build our frontend Next project, due to an invalid project directory being provided (despite it not having changed).

`error - Invalid project directory provided, no such directory: /Users/myusername/workspace/project-name/apps/frontend/apps/frontend`

Having played with the project config to try and resolve it, it looks like `nx/next` (the build executor) is appending the project `sourceRoot` twice.

For context, the project.json for the frontend application looks like this:

```{
  "name": "frontend",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "apps/frontend",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nx/next:build",
      "outputs": [
        "{options.outputPath}"
      ],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "dist/apps/frontend"
      },
      "configurations": {
        "development": {
          "outputPath": "apps/frontend"
        },
        "production": {}
      }
    },
    "serve": {
      "executor": "@nx/next:server",
      "defaultConfiguration": "development",
      "options": {
        "buildTarget": "frontend:build",
        "dev": true
      },
      "configurations": {
        "development": {
          "buildTarget": "frontend:build:development",
          "dev": true
        },
        "production": {
          "buildTarget": "frontend:build:production",
          "dev": false
        }
      }
    },
    "export": {
      "executor": "@nx/next:export",
      "options": {
        "buildTarget": "frontend:build:production"
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "apps/frontend/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "apps/frontend/**/*.{ts,tsx,js,jsx}"
        ]
      }
    }
  },
  "tags": []
}
```

This specifically seems to relate to projects using yarn berry.

### Expected Behavior

I would expect the project to build as it does in previous versions of nx (including 16.6.x) when `yarn nx run frontend:build:production` is executed.

### GitHub Repo

https://github.com/Neverbland/nx-issue

### Steps to Reproduce

1. Clone the GitHub repo
2. `cd` into the project directory
3. Run `yarn`
4. Run `yarn nx run  frontend:build:production`

### Nx Report

```shell
   Node   : 16.16.0
   OS     : darwin-arm64
   yarn   : 3.6.1

   nx                 : 16.7.0
   @nx/js             : 16.7.0
   @nx/jest           : 16.7.0
   @nx/linter         : 16.7.0
   @nx/workspace      : 16.7.0
   @nx/devkit         : 16.7.0
   @nx/eslint-plugin  : 16.7.0
   @nx/next           : 16.7.0
   @nx/react          : 16.7.0
   @nrwl/tao          : 16.7.0
   @nx/web            : 16.7.0
   typescript         : 5.1.6
```

### Failure Logs

```shell
yarn nx run  frontend:build:production

> nx run frontend:build:production

- error Invalid project directory provided, no such directory: /Users/alain/workspace/nx-issue/apps/frontend/apps/frontend
Error occurred while trying to run the yarn next build apps/frontend
Error: Command failed: yarn next build apps/frontend
    at checkExecSyncError (node:child_process:828:11)
    at execSync (node:child_process:899:15)
    at /Users/alainritchie/workspace/nx-issue/node_modules/@nx/next/src/executors/build/build.impl.js:54:42
    at Generator.next (<anonymous>)
    at /Users/alainritchie/workspace/nx-issue/node_modules/tslib/tslib.js:169:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/alainritchie/workspace/nx-issue/node_modules/tslib/tslib.js:165:16)
    at buildExecutor (/Users/alainritchie/workspace/nx-issue/node_modules/@nx/next/src/executors/build/build.impl.js:20:20)
    at /Users/alainritchie/workspace/nx-issue/node_modules/nx/src/command-line/run/run.js:81:23

 —————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project frontend (1s)

    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```

### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

We believe this issue was introduced as part of commit [cd6c8d3](https://github.com/nrwl/nx/commit/cd6c8d373a7f882b45d50f79eaa8a915ef00be9e)

Specifically this:

``` 
const buildCommand = isYarnBerry
    ? `yarn next build ${projectRoot}`
    : 'npx next build';
```

and is fixed with this simple change to `packages/next/src/executors/build/build.impl.ts`
``` 
const buildCommand = isYarnBerry
    ? `yarn next build`
    : 'npx next build';
```

### PR DESCRIPTION
**Pull request by ndcunningham**
What has changed:

Instead of running different build commands based on the package manager we can invoke the command directly from the bin.

closed #18653

### CODE REVIEW

File: nx-dev/data-access-ai/src/lib/data-access-ai.ts [hunk 1]
```diff
@@ -124,6 +124,8 @@ export async function queryAi(
       );
     }
 
+    console.log({ embeddingResponse });
```
Comment (meeroslav) [inline nx-dev/data-access-ai/src/lib/data-access-ai.ts:127]: Please remove these

File: nx-dev/data-access-ai/src/lib/data-access-ai.ts [hunk 2]
```diff
@@ -137,6 +139,7 @@ export async function queryAi(
         min_content_length: MIN_CONTENT_LENGTH,
       }
     );
+    console.log({ pageSections });
```
Comment (meeroslav) [inline nx-dev/data-access-ai/src/lib/data-access-ai.ts:142]: Please remove this

File: nx-dev/data-access-ai/src/lib/utils.ts [hunk 3]
```diff
@@ -19,6 +19,12 @@ export function getMessageFromResponse(
   return response.choices[0].message?.content ?? '';
 }
 
+// Improve this function so that urls are not repeated
+
+// export function getHeadingUrlfromPageSection(pageSections: PageSection[]): {
+//   heading: string;
+//   url: string;
+// } {}
```
Comment (meeroslav) [inline nx-dev/data-access-ai/src/lib/utils.ts:27]: Please remove this

File: nx-dev/feature-ai/src/lib/feed-container.tsx [hunk 4]
```diff
@@ -65,6 +65,8 @@ export function FeedContainer(): JSX.Element {
         query,
         lastAnswerChatItem ? lastAnswerChatItem.content : ''
       );
+
+      console.log({ aiResponse });
```
Comment (meeroslav) [inline nx-dev/feature-ai/src/lib/feed-container.tsx:69]: Please remove this

File: packages/next/src/executors/build/build.impl.ts [hunk 5]
```diff
@@ -49,32 +49,20 @@ export default async function buildExecutor(
     process.env['__NEXT_REACT_ROOT'] ||= 'true';
   }
 
-  const { experimentalAppOnly, profile, debug, outputPath } = options;
-
   // Set output path here since it can also be set via CLI
   // We can retrieve it inside plugins/with-nx
-  process.env.NX_NEXT_OUTPUT_PATH ??= outputPath;
+  process.env.NX_NEXT_OUTPUT_PATH ??= options.outputPath;
```
Comment (meeroslav) [inline packages/next/src/executors/build/build.impl.ts:54]: Why not set this also in the `runCliBuild` since we already extract the `outputPath` there?

**Review by meeroslav — CHANGES_REQUESTED** [review]
Few minor changes needed, otherwise it's good.

File: packages/next/src/executors/build/build.impl.ts (hunk 5 above)
Comment (ndcunningham) [reply packages/next/src/executors/build/build.impl.ts:54]: I could it has nothing to do with the CLI build itself though

**Review by meeroslav — APPROVED** [review]

**Review by xiongemi — APPROVED** [review]

**Pull request merged**